File: chatserve.c
/*********************************************************************
** Author: James G Wilson
** Date: 5/1/2018
** Description: chatserve.c
*********************************************************************/
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <sys/socket.h>
#include <netdb.h>
#include <string.h>

void printUsage();

int main(int argc, char*argv[])
{
	
	/* Check command line bad usage.
	 * Only checks if the number of arguments is wrong. */
	if(argc != 2)
	{
		printUsage();
	}

	// Get port number from command line arg
	// Wait on that port for a client request
	
	// Taken from Beej's Guide to Network Programming :D
	// Set up server to listen on this host's IP address, port ####
	int status;
	struct addrinfo hints;		// fill out with relevent info
	struct addrinfo *servinfo;	// will point to results

	memset(&hints, 0, sizeof(hints));	// make sure the struct is empty
	hints.ai_family = AF_UNSPEC;		// don't care IPv4 or IPv6
	hints.ai_socktype = SOCK_STREAM;	// TCP stream sockets (woot!)
	hints.ai_flags = AI_PASSIVE;		// fill in my IP for me
	
	if((status = getaddrinfo(NULL, "3490", &hints, &servinfo)) != 0)
	{
		fprintf(stderr, "getaddrinfo error: %s\n", gai_strerror(status));
		exit(1);
	}

	// servinfo now points to a linked list of 1 or more struct addrinfos
	
	// . . . do everything until you don't need servinfo anymore . . . .
	
	freeaddrinfo(servinfo); // free the linked-list

	return 0;
}

/* Print command line argument format */
void printUsage()
{ 
	printf("Usage ./chatserve <port#>\n");
}

File: chatserve.py
#/*********************************************************************
#** Author: James G Wilson
#** Date: 5/4/2018
#** Description: chatserve.py 
#*********************************************************************/

# Access socket methods without having to prepend 'socket.'
from socket import * 

def main():

	# Hard code server's handle.
	# handle = 'Host A (server)'
	# print(handle)

	# get port # from command line and convert to int
	serverPort = input('Enter server port#: ')
	# print('You entered ' + serverPort)

	# create IPv6 TCP listening socket
	serverSocket = socket(AF_INET6, SOCK_STREAM)

	# bind port # with listening socket
	serverSocket.bind(('', int(serverPort)))

	# start listening with backlog = 1
	serverSocket.listen(1)
	print('Server is now listening and ready to receive...')

	# wait, indefinitely, for client to come knocking
	while True:

		# addr is address bound to the (client's)
		# socket on the other end of the connection
		connectionSocket, addr = serverSocket.accept()

		# Establish connection with client.
		clientMessage = connectionSocket.recv(1024);
		print(clientMessage.decode())

		#message = 'Server connection established.'
		#connectionSocket.send(message.encode())

if __name__ == "__main__":
	# execute only if run as a script
	main()


